src/index.ts

```diff
@@ -379,6 +402,75 @@ class LoopSDK {
       },
     };
   }
+
+  private getCachedConnection(): any | null {
+    const existingConnectionRaw = typeof window !== 'undefined' ? localStorage.getItem('loop_connect') : null;
```

Comment:
these typeof window check feel redundant.  we probably should just do this check once when initialize SDK and throrw error right there. by the time we enter these realm we should had localStorage .

Or a better check is checking localStorage support early on and throw error if not support local storage instead.